### PR TITLE
Read a paragraph-per-item requirements section, not just a list (reqextract)

### PR DESCRIPTION
**Pull request by strelov1**
## Summary

- `reqextract.Derive` only read requirement items out of a `<ul>`/`<ol>` list under a recognized heading; a section stated as a run of `<p>` items closed as prose before any list was found and yielded nothing. Not hypothetical: every one of 683 real `Требования` headings sampled from `tbank.ru` is followed by a run of `<p>` paragraphs, never a list — so the Russian heading vocabulary added in `reqextract-russian-headings` matches the heading on that source today and still extracted nothing from it.
- Added a deferred buffer (`pending`/`pendingHasProse`/`flushPending`) to `Derive`'s walk: a too-long-for-a-heading text block is buffered instead of closing its section immediately, since it may be the section's own first stated item rather than prose explaining the section away — the two cannot be told apart from one paragraph alone. A list found afterward discards the buffer only when real prose preceded it (`pendingHasProse`); a heading transition, a table, or the document's end commits the buffer's paragraphs as the section's items, but only when there are two or more of them.
- Updated `internal/job/reqextract/AGENTS.md`: "How it works" describes the new mechanism, the now-fixed `<p>`-per-item Limitations bullet is removed, and the "Measuring coverage" figures (28.0%/29.3%) are flagged as predating this fix.
- OpenSpec artifacts under `openspec/changes/reqextract-paragraph-items/` (proposal, design, spec delta, tasks — all tasks checked off).

## Test plan

- [x] `go test ./internal/job/reqextract/... -v -run TestDerive` — every pre-existing case unchanged, the updated `tbank.ru`-shape negative test now asserts the derived requirements, and 7 new boundary-case tests (exactly-two-items commit, single-paragraph-yields-nothing, mixed short/long paragraphs, preferred-section paragraphs, non-vocabulary heading yields nothing, no-trailing-heading commit, table-closed commit) all pass.
- [x] Verified against 25 real `tbank.ru` posting descriptions (sampled live, `Требования` heading, paragraph-per-item shape) run through the actual `Derive` function — every one yielded the expected 3-10 requirement items, none yielded zero.
- [x] `gofmt -l .` clean, `go build ./...`, `go vet ./...`, `go vet -tags=integration ./...` all clean.
- [x] `go test ./...` (whole repo, 214 packages) — 0 FAIL.
- [x] `openspec validate reqextract-paragraph-items --strict` passes.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Requirements written as multiple paragraphs are now extracted as individual items alongside list-based requirements.
  * Required and preferred priorities are preserved for paragraph-based sections.

* **Bug Fixes**
  * Fixed paragraph-based requirement sections incorrectly producing no results.
  * Improved handling of section boundaries, tables, and document endings.
  * Prevented explanatory prose, single paragraphs, and nested formatting text from being misclassified as requirements.

### CODE REVIEW

File: openspec/changes/reqextract-paragraph-items/design.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+**`pendingHasProse`, not "any pending buffer discards on a list".** A short
+unrecognized inline line (a lead-in like `<p>You will need:</p>`, or a bare
+spacer `<p></p>`) must NOT prevent a following list from being read — that
+behavior already existed before this change (the walk already tolerated
+lead-in lines between a heading and its list) and regressing it would break
+real postings that use exactly that shape. Only a line long enough to have
+failed the heading-length test (real prose) should cause a following list to
+be treated as belonging to a DIFFERENT section. `pendingHasProse` is the one
+bit that keeps these two apart: set only by the over-length branch, read only
+by the list-found branch.
```
[the remark this answers is not in the document]
Comment (strelov1) [reply openspec/changes/reqextract-paragraph-items/design.md:78]: Verified against the actual code rather than applying: `headingDecision` already returns `("", true)` for an unrecognized h1–h6 (the switch's default case at the bottom of the function), so `Derive` takes the `recognized` branch, calls `flushPending(priority)`, and sets `priority = ""` — it does not fall into the buffering branch this finding describes. Reproduced with the exact scenario proposed here: `Derive(`<h3>Requirements</h3><h3>Benefits</h3><ul><li>Free lunch</li></ul>`)` returns zero entries on the current code (verified both before and after the sibling fix in this PR). No change made — this appears to be a false positive.
